bundles/com.espressif.idf.ui/src/com/espressif/idf/ui/tools/ToolsJob.java

```diff
@@ -349,6 +349,11 @@ protected IStatus runCommandIdfPyInIdfEnv(List<String> arguments, MessageConsole
 			Logger.log(environment.toString());
 			environment.put("PYTHONUNBUFFERED", "1"); //$NON-NLS-1$ //$NON-NLS-2$
 			loadIdfPathWithSystemPath(environment);
+			environment.remove(IDFEnvironmentVariables.IDF_PYTHON_ENV_PATH);
+			environment.remove(IDFEnvironmentVariables.ESP_IDF_VERSION);
+			environment.remove(IDFEnvironmentVariables.IDF_PATH);
+			environment.remove(IDFEnvironmentVariables.OPENOCD_SCRIPTS);
```

Comment:
Hi @alirana01, 

Do you know why these variables are not replaced with the same ones from the idfToolSet (which, as I understand, should have proper environment variables)?

Reply:
So basically the environment here is taken from `System.getEnv()` so it loads the system env. The function where the change was done is actually before these variables are even part of the idfToolSet. This function make sures that the correct variables from the idf export script are loaded into the idfToolSet basically it sits before that. Which is the reason we need to make sure that any of these variables from System environment are not loaded here